### PR TITLE
Functions for user-driven tuning

### PR DESCRIPTION
**Pull request by benjaminulmer**
This PR adds the two functions needed in Tensile to enable user-driven tuning in rocBLAS

1. MasterSolutionLibrary::getSolutionByIndex
2. ContractionSolution::canSolve

getSolutionByIndex only returns loaded solutions; the caller (rocBLAS for now) is responsible for making sure the solution is already loaded

### CODE REVIEW

**Review by AlexBrownAMD — COMMENTED** [review]
Is the rest of the user-driven tuning process driven by the rocblas library? How are library logic updates handled?

**Comment by benjaminulmer** [comment]
> Is the rest of the user-driven tuning process driven by the rocblas library? How are library logic updates handled?

Yes, rocBLAS handles the rest. For context, here's what's being added to rocBLAS:
1. New rocblas_gemm<strided+batched>_ex_get_solutions​ functions that will return all solutions that can solve a given GEMM
2. Making the solution_index​ parameter of the gemm_ex functions functional 
3. New rocblas_gemm_flags_check_solution_index​ flag that checks if a solution can solve a problem; does not run the solution
4. New rocblas_status_invalid_solution_index​ status for when a solution cannot solve a problem 

The user is responsible for benchmarking solutions and tracking which solution is best for which problem, so the actual library logic is not updated.

File: Tensile/Source/lib/include/Tensile/MasterSolutionLibrary.hpp
Comment (benjaminulmer) [reply Tensile/Source/lib/include/Tensile/MasterSolutionLibrary.hpp]: added null check

**Comment by nakajee** [comment]
Thanks for your update. I do not have any further comment. Waiting for comments/approval from others.

**Review by AlexBrownAMD — APPROVED** [review]

**Review by nakajee — APPROVED** [review]

**Pull request merged**
